src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc

```diff
@@ -0,0 +1,526 @@
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest, OnEvent_IncorrectFunctionId) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MessageSharedPtr msg_hmi_response = CreateFullParamsSO();
+
+  Event event_ui(hmi_apis::FunctionID::UI_PerformAudioPassThru);
+  event_ui.set_smart_object(*msg_hmi_response);
+
+  EXPECT_CALL(mock_message_helper_, HMIToMobileResult(_)).Times(0);
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL))
+      .Times(0);
+
+  command->on_event(event_ui);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       OnEvent_HMIResponseSUCCESS_SUCCESS) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MessageSharedPtr msg_hmi_response = CreateFullParamsSO();
+  (*msg_hmi_response)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::SUCCESS;
+
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+  const mobile_apis::Result::eType mobile_response =
+      mobile_apis::Result::SUCCESS;
+  const bool is_success = kIsSuccess;
+
+  MediaAppSetup();
+  DefineHMILevelUIAvailable();
+  ON_CALL(*mock_app_, AddIndicatorWaitForResponse(_))
+      .WillByDefault(Return(true));
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_));
+
+  command->Run();
+
+  EXPECT_CALL(*mock_app_, set_audio_streaming_indicator(_));
+  EXPECT_CALL(*mock_app_, RemoveIndicatorWaitForResponse(_));
+  TestCommandOnEventCall(msg_mobile_response,
+                         mobile_response,
+                         is_success,
+                         msg_hmi_response,
+                         command);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       OnEvent_HMIResponseUNSUPPORTED_RESOURCE_SUCCESS) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MessageSharedPtr msg_hmi_response = CreateFullParamsSO();
+  (*msg_hmi_response)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE;
+
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+  const mobile_apis::Result::eType mobile_response =
+      mobile_apis::Result::UNSUPPORTED_RESOURCE;
+  const bool is_success = kIsSuccess;
+
+  MediaAppSetup();
+  DefineHMILevelUIAvailable();
+  ON_CALL(*mock_app_, AddIndicatorWaitForResponse(_))
+      .WillByDefault(Return(true));
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_));
+
+  command->Run();
+
+  EXPECT_CALL(*mock_app_, set_audio_streaming_indicator(_));
+  EXPECT_CALL(*mock_app_, RemoveIndicatorWaitForResponse(_));
```

Comment:
@tkireva, are we really don't interested in arguments value?
Please, check everywhere